### PR TITLE
Improve peers logic, fixed SQL error during peers dbSave

### PR DESCRIPTION
**Pull request by 4miners**

### CODE REVIEW

File: logic/peer.js [hunk 1]
```diff
@@ -45,42 +51,59 @@ Peer.prototype.nullable = [
...
+			this[key] = peer[key];
+		}
+	}.bind(this));
+
+	// Adjust properties according to rules
+	if (/^[0-9]+$/.test(this.ip)) {
+		this.ip = ip.fromLong(this.ip);
 	} else {
-		this.ip = peer.ip;
+		this.ip = this.ip;
```
Comment (MaciejBaj) [inline logic/peer.js]: ?

File: logic/peer.js [hunk 2]
```diff
@@ -45,42 +51,59 @@ Peer.prototype.nullable = [
...
-		this.state = 1;
-	}
+	return this;
+};
 
+Peer.prototype.normalize = function (peer) {
 	if (peer.dappid) {
 		if (Array.isArray(peer.dappid)) {
-			this.dappid = peer.dappid;
+			peer.dappid = peer.dappid;
```
Comment (MaciejBaj) [inline logic/peer.js]: by accident?

File: logic/peer.js (hunk 1 above)
Comment (4miners) [inline logic/peer.js]: Yep, thanks.

File: logic/peer.js (hunk 2 above)
Comment (4miners) [inline logic/peer.js]: Yep, thanks.

File: logic/peer.js [hunk 3]
```diff
@@ -92,29 +106,19 @@ Peer.prototype.parseInt = function (integer, fallback) {
...
-	_.each(object, function (value, key) {
+Peer.prototype.update = function (peer) {
+	peer = self.normalize(peer);
+
+	// Accept only supported properties
+	_.each(self.properties, function (key) {
 		// Change value only when is defined, also prevent release ban when banned peer connect to our node
-		if (value !== null && value !== undefined && !(key === 'state' && this.state === 0 && object.state === 2)) {
-			this[key] = value;
+		if (peer[key] !== null && peer[key] !== undefined && !(key === 'state' && this.state === 0 && peer.state === 2) && !_.includes(self.immutable, key)) {
```
Comment (MaciejBaj) [inline logic/peer.js]: so we cannot unset properties that were set before.

Comment (MaciejBaj) [inline logic/peer.js]: are these 2 checks needed?

File: logic/peer.js [hunk 4]
```diff
@@ -45,42 +51,50 @@ Peer.prototype.nullable = [
...
-		this.ip = ip.fromLong(peer.ip);
-	} else {
-		this.ip = peer.ip;
-	}
+	// Normalize peer data
+	peer = self.normalize(peer);
+
+	// Accept only supported and defined properties
+	_.each(self.properties, function (key) {
+		if (peer[key] !== null && peer[key] !== undefined) {
```
Comment (MaciejBaj) [inline logic/peer.js]: also here, why we won't let to unset properties?

Comment (MaciejBaj) [inline logic/peer.js]: ok as we discussed- let's leave it as it is

File: logic/peer.js (hunk 3 above)
Comment (MaciejBaj) [inline logic/peer.js]: ok as we discussed- let's leave it as it is

**Pull request merged**